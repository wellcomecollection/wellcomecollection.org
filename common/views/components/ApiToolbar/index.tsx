import { FunctionComponent, useState } from 'react';
import styled from 'styled-components';
import cookies from '@weco/common/data/cookies';
import { getCookie, setCookie } from 'cookies-next';
import useIsomorphicLayoutEffect from '../../../hooks/useIsomorphicLayoutEffect';
import { Contributor, License } from '../../../model/catalogue';

export type ApiToolbarLink = {
  id: string;
  label: string;
  value?: string;
  link?: string;
};

const ToolbarContainer = styled.div<{ mini: boolean }>`
  display: ${props => (props.mini ? 'inline-block' : 'flex')};
  background-color: ${props => props.theme.color('accent.purple')};
  color: ${props => props.theme.color('white')};
  z-index: 100;
`;

const LinkList = styled.ul.attrs({
  className: 'font-size-5',
})`
  display: flex;
  list-style: none;
  margin: 0;
`;

/** tzitzit is a tool used by the Digital Editorial team to create TASL
 * strings for Prismic. https://github.com/wellcomecollection/tzitzit
 *
 * This creates a link to a tzitzit which is pre-filled with the catalogue
 * metadata for this image.  They may not want to use it directly
 * (e.g. they may tidy up the title), but it should save them some copy/pasting
 * from the catalogue.
 *
 */

export function setTzitzitParams({
  title,
  sourceLink,
  licence,
  contributors,
}: {
  title: string;
  sourceLink: string;
  licence: License | undefined;
  contributors: Contributor[] | undefined;
}): ApiToolbarLink | undefined {
  const licenceId = licence?.id.toUpperCase();

  // We should not be using in copyright images in Stories.
  // TODO: Double check this is our policy with the Editorial team.
  if (licenceId === 'INC')
    return {
      id: 'tzitzit',
      label: '(no tzitzit, this image is in copyright)',
    };

  const params = new URLSearchParams();
  params.set('title', title);
  params.set('sourceName', 'Wellcome Collection');
  params.set('sourceLink', sourceLink);
  if (licenceId) params.set('licence', licenceId);
  if (contributors && contributors.length > 0)
    params.set('author', contributors[0].agent.label);

  return {
    id: 'tzitzit',
    label: 'tzitzit',
    link: `https://s3-eu-west-1.amazonaws.com/tzitzit.wellcomecollection.org/index.html?${params.toString()}`,
  };
}

export function createPrismicLink(prismicId: string): ApiToolbarLink {
  return {
    id: 'prismic',
    label: 'Edit in Prismic',
    link: `https://wellcomecollection.prismic.io/documents~b=working&c=published&l=en-gb/${prismicId}/`,
  };
}

function getAnchorLinkUrls() {
  // This function currently only extracts the ids from h2, h3, and h4 tags
  const getAllHeadingIds = [...document.querySelectorAll('h2, h3, h4')].map(
    item => item.id
  );
  // This function extracts any data-toolbar-anchor divs with the view to extracting the id values
  // This can be used across the codebase (where data-toolbar-anchor is used) but at the moment is only used in audio & BSL guides
  // Please note: an audio/BSL guide must contain and audio or video file with an accompanying title or no id will exist
  // and no link will be created
  const extractedAudioBSLAttributes = [
    ...document.querySelectorAll('div[data-toolbar-anchor="apiToolbar"]'),
  ].map(item => item.id);

  // Remove empty ids and then append them to the current url with # to
  // create the anchor link
  // e.g. weco.org/guides/exhibitions/YvUALRAAACMA2h8V/captions-and-transcripts#anchor-id
  const extractedHeadingIdURLs = getAllHeadingIds
    .filter(Boolean)
    .map(id => `${document.URL}#${id}`);
  const extractedAudioBSLURLs = extractedAudioBSLAttributes
    .filter(Boolean)
    .map(id => `${document.URL}#${id}`);

  const csvAsSingleColumn =
    extractedHeadingIdURLs.join('\n') + extractedAudioBSLURLs.join('\n');
  // Push the list of urls to the clipboard
  if (navigator && navigator.clipboard && navigator.clipboard.writeText)
    return navigator.clipboard.writeText(csvAsSingleColumn).then(() => {
      alert('All anchor links on this page have been copied to clipboard!');
    });
  return alert(
    'The Clipboard API is not available. No anchor links have been copied.'
  );
}

type Props = {
  links?: ApiToolbarLink[];
};

const ApiToolbar: FunctionComponent<Props> = ({ links = [] }) => {
  const [mini, setMini] = useState<boolean>(false);

  useIsomorphicLayoutEffect(() => {
    setMini(getCookie(cookies.apiToolbarMini) === true);
  }, []);

  const propValue = (prop: ApiToolbarLink) => {
    return `${prop.label}${prop.value ? ` : ${prop.value}` : ''}`;
  };

  return (
    <ToolbarContainer mini={mini}>
      <div
        style={{
          display: 'flex',
          alignItems: 'center',
          flexGrow: 1,
        }}
      >
        {!mini && (
          <>
            <span
              className="h3"
              style={{
                marginLeft: '10px',
              }}
            >
              API toolbar
            </span>
            <LinkList>
              {links.map(prop => (
                <li
                  key={prop.id}
                  style={{
                    paddingRight: '10px',
                    paddingLeft: '10px',
                    borderLeft: '1px solid #bcbab5',
                  }}
                >
                  {prop.link && <a href={prop.link}>{propValue(prop)}</a>}
                  {!prop.link && propValue(prop)}
                </li>
              ))}
            </LinkList>
          </>
        )}
      </div>
      <button
        className="plain-button"
        type="button"
        onClick={() => {
          getAnchorLinkUrls();
        }}
        style={{ padding: '10px' }}
      >
        <>‚öìÔ∏è</>
      </button>

      <button
        className="plain-button"
        type="button"
        onClick={() => {
          setMini(!mini);
          setCookie(cookies.apiToolbarMini, !mini, { path: '/' });
        }}
        style={{ padding: '10px' }}
      >
        {!mini && <>ü§è</>}
        {mini && <>üëê</>}
      </button>
    </ToolbarContainer>
  );
};

export default ApiToolbar;
