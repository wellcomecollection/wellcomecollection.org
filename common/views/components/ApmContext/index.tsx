import { ApmBase } from '@elastic/apm-rum';
import getConfig from 'next/config';
import { createContext, FunctionComponent, PropsWithChildren } from 'react';

import useApmRum from './useApmRum';

type ApmContextData = {
  apm?: ApmBase;
};

export const ApmContext = createContext<ApmContextData>({});

const { publicRuntimeConfig } = getConfig();

export const ApmContextProvider: FunctionComponent<PropsWithChildren> = ({
  children,
}) => {
  const apm = useApmRum(publicRuntimeConfig.apmConfig);
  return <ApmContext.Provider value={{ apm }}>{children}</ApmContext.Provider>;
};
