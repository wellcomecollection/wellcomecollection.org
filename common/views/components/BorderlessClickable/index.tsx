import {
  ComponentProps,
  forwardRef,
  ForwardRefRenderFunction,
  ReactNode,
  SyntheticEvent,
} from 'react';
import styled from 'styled-components';

import { IconSvg } from '@weco/common/icons';
import { classNames, font } from '@weco/common/utils/classnames';
import {
  BaseButtonInner,
  BasicButton,
  ButtonIconWrapper,
} from '@weco/common/views/components/Buttons/Buttons.styles';
import Icon from '@weco/common/views/components/Icon';

type ClickableElement = 'a' | 'button';

type StyleProps = {
  $isActive?: boolean;
};

const BorderlessClickableStyle = styled(BasicButton)<StyleProps>`
  background: transparent;
  color: ${props => props.theme.color('neutral.700')};
  padding: 10px 8px;

  &:hover {
    background: ${props => props.theme.color('neutral.300')};
  }

  ${props =>
    props.$isActive &&
    `
    background: ${props.theme.color('neutral.300')};
  `}
`;

type Props = {
  icon?: IconSvg;
  iconLeft?: IconSvg;
  text: ReactNode;
  isTextHidden?: boolean;
  isActive?: boolean;
};

type BorderlessClickableProps = Props & { as: ClickableElement };
const Button: ForwardRefRenderFunction<
  HTMLButtonElement,
  BorderlessClickableProps
> = (
  {
    as,
    icon,
    iconLeft,
    text,
    isTextHidden,
    isActive,
    ...elementProps
  }: BorderlessClickableProps,
  ref
) => {
  return (
    <BorderlessClickableStyle
      as={as}
      $isActive={isActive}
      ref={ref}
      {...elementProps}
    >
      <BaseButtonInner $isInline={true}>
        <>
          {iconLeft && (
            <ButtonIconWrapper $iconAfter={false}>
              {/* This is all a little hacky and will need some tidy up */}
              {/* We currently only use this in the header sign in button */}
              <span
                className={font('intr', 4)}
                style={{ transform: 'translateY(0.01em)' }}
              >
                <Icon icon={iconLeft} matchText={true} />
              </span>
            </ButtonIconWrapper>
          )}
          <span
            className={classNames({
              'visually-hidden': !!isTextHidden,
            })}
          >
            {text}
          </span>
          {icon && (
            <ButtonIconWrapper $iconAfter={true}>
              <Icon icon={icon} />
            </ButtonIconWrapper>
          )}
        </>
      </BaseButtonInner>
    </BorderlessClickableStyle>
  );
};

const BorderlessClickable = forwardRef<
  HTMLButtonElement,
  BorderlessClickableProps
>(Button);

type BorderlessLinkProps = Props & ComponentProps<'a'>;
const Link = (props, ref) => {
  return <BorderlessClickable as="a" ref={ref} {...props} />;
};

const BorderlessLink = forwardRef<HTMLButtonElement, BorderlessLinkProps>(Link);

type BorderlessButtonProps = Props &
  ComponentProps<'button'> & {
    clickHandler?: (event: SyntheticEvent<HTMLButtonElement>) => void;
  };
const ButtonOuter = (
  { clickHandler, ...elementProps }: BorderlessButtonProps,
  ref
) => {
  function onClick(event: SyntheticEvent<HTMLButtonElement>) {
    clickHandler && clickHandler(event);
  }

  return (
    <BorderlessClickable
      as="button"
      onClick={onClick}
      ref={ref}
      {...elementProps}
    />
  );
};

const BorderlessButton = forwardRef<HTMLButtonElement, BorderlessButtonProps>(
  ButtonOuter
);

export { BorderlessLink, BorderlessButton };
