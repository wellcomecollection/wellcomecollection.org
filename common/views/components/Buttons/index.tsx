import {
  ForwardRefRenderFunction,
  ForwardedRef,
  PropsWithChildren,
  forwardRef,
} from 'react';
import ButtonSolid, { ButtonSolidProps } from './Buttons.Solid';
import ButtonSolidLink, { ButtonSolidLinkProps } from './Buttons.SolidLink';
import DropdownButton, { DropdownButtonProps } from './Buttons.Dropdown';
// import {
//   BorderlessLink,
//   BorderlessButton,
//   BorderlessLinkProps,
//   BorderlessButtonProps,
// } from './Buttons.Borderless';

type ButtonProps =
  // | (BorderlessButtonProps & {
  //     variant: 'BorderlessButton';
  //   })
  // | (BorderlessLinkProps & {
  //     variant: 'BorderlessLink';
  //   })
  | (PropsWithChildren<DropdownButtonProps> & {
      variant: 'DropdownButton';
    })
  | (ButtonSolidProps & {
      variant: 'ButtonSolid';
    })
  | (ButtonSolidLinkProps & {
      variant: 'ButtonSolidLink';
    });

const Button: ForwardRefRenderFunction<HTMLButtonElement, ButtonProps> = (
  props: ButtonProps,
  ref?: ForwardedRef<HTMLButtonElement>
) => {
  const { variant } = props;

  switch (variant) {
    // case 'BorderlessButton':
    //   return <BorderlessButton {...props} />;
    // case 'BorderlessLink':
    //   return <BorderlessLink {...props} />;
    case 'DropdownButton':
      return <DropdownButton {...props} />;
    case 'ButtonSolidLink':
      return <ButtonSolidLink {...props} />;
    case 'ButtonSolid':
    default:
      return <ButtonSolid ref={ref} {...props} />;
  }
};

export default forwardRef<HTMLButtonElement, ButtonProps>(Button);
export * from './Buttons.styles';
export * from './Buttons.types';
