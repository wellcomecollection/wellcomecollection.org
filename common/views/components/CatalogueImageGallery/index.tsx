import { FunctionComponent } from 'react';

import CatalogueImageGalleryJustified, {
  Props as CatalogueImageGalleryJustifiedProps,
} from '@weco/common/views/components/CatalogueImageGallery/CatalogueImageGallery.Justified';
import CatalogueImageGalleryScrollable, {
  Props as CatalogueImageGalleryScrollableProps,
} from '@weco/common/views/components/CatalogueImageGallery/CatalogueImageGallery.Scrollable';

type Props =
  | (CatalogueImageGalleryJustifiedProps & { variant: 'justified' })
  | (CatalogueImageGalleryScrollableProps & { variant: 'scrollable' });

const CatalogueImageGallery: FunctionComponent<Props> = props => {
  const { variant } = props;

  return (
    <>
      {variant === 'justified' ? (
        <CatalogueImageGalleryJustified {...props} />
      ) : (
        <CatalogueImageGalleryScrollable {...props} />
      )}
    </>
  );
};

export default CatalogueImageGallery;
