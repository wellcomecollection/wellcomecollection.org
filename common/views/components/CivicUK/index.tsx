const branding = {
  removeIcon: true,
  removeAbout: true,
  fontFamily: 'Inter, sans-serif',
};

const text = {
  title: 'Manage cookies',
  intro: '',
  necessaryTitle: 'Essential cookies',
  necessaryDescription:
    'These cookies are necessary for our website to function and therefore always need to be on.',
  notifyTitle: 'Our website uses cookies',
  notifyDescription:
    "We use cookies to make our website work. To help us make our marketing and website better, we'd like your consent to use cookies on behalf of third parties too.",
  closeLabel: 'Save and close',
  settings: 'Manage cookies',
  accept: 'Accept all',
  acceptSettings: 'Accept all',
  reject: 'Essential only',
  rejectSettings: 'Essential only',
};

// Should your privacy policy change after a user gives consent,
// Cookie Control will invalidate prior records of consent and seek the user's preferences using the latest information available.
// https://www.civicuk.com/cookie-control/documentation/cookies
const statement = {
  description:
    "We use cookies to make our website work. To help us make our marketing and website better, we'd like your consent to use cookies on behalf of third parties too.",
  name: 'Find out more in our privacy policy',
  url: 'https://wellcome.org/who-we-are/privacy-and-terms',
  updated: '25/05/2018',
};

const CivicUK = () => (
  <>
    <script
      src="https://cc.cdn.civiccomputing.com/9/cookieControl-9.x.min.js"
      type="text/javascript"
    ></script>
    <script
      dangerouslySetInnerHTML={{
        __html: `CookieControl.load({
            product: 'COMMUNITY',
            apiKey: '5a1b492f47fac442240976a4f991ad92278ef15a',
            product: 'pro',
            initialState: 'notify',
            layout: 'popup',
            setInnerHTML: true,
            closeStyle: 'button',
            settingsStyle: 'button',
            notifyDismissButton: false,
            necessaryCookies: ['toggle_*'],
            optionalCookies: [
              {
                name: 'analytics',
                label: 'Measure website use',
                description:
                  '<ul><li>We use these cookies to recognise you, to count your visits to the website, and to see how you move around it.</li><li>They help us to provide you with a good experience while you browse, for example by helping to make sure you can find what you need.</li><li>They also allows us to improve the way the website works.</li></ul>',
                cookies: [
                  '_ga',
                  '_ga*',
                  '_gid',
                  '_gat',
                  '__utma',
                  '__utmt',
                  '__utmb',
                  '__utmc',
                  '__utmz',
                  '__utmv',
                ],
                onAccept: function () {
                  const event = new CustomEvent('analyticsConsentChanged', { detail: { consent: 'granted' }});
                  window.dispatchEvent(event);
                },
                onRevoke: function () {
                  const event = new CustomEvent('analyticsConsentChanged', { detail: { consent: 'denied' } });
                  window.dispatchEvent(event);
                },
              },
            ],   
            statement: ${JSON.stringify(statement)},
            branding: ${JSON.stringify(branding)},
            text: ${JSON.stringify(text)}
          });`,
      }}
    />
  </>
);

export default CivicUK;
