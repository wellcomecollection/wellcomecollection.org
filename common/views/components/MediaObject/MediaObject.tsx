import { FunctionComponent, ReactElement } from 'react';
import Image from '../Image/Image';
import MediaObjectBase, {
  TextWrapperProp,
} from '../MediaObjectBase/MediaObjectBase';
import { ImageType } from '../../../model/image';
import PrismicHtmlBlock from '../PrismicHtmlBlock/PrismicHtmlBlock';
import { HTMLString } from '@weco/common/services/prismic/types';
import styled from 'styled-components';
import { grid, classNames, font } from '../../../utils/classnames';

type Props = {
  title: string;
  text: HTMLString | null;
  image: ImageType;
};

type ImageWrapperProp = {
  hasImage: boolean;
};

const grid12 = grid({ s: 12, m: 12, l: 12, xl: 12 });

const ImageWrapper = styled.div.attrs<ImageWrapperProp>(props => {
  if (props.hasImage) {
    return {
      className: grid({ s: 2, m: 2, l: 2, xl: 2 }),
    };
  }
  return {
    className: grid12,
  };
})<ImageWrapperProp>``;

const TextWrapper = styled.div.attrs<TextWrapperProp>(props => {
  if (props.hasImage) {
    return {
      className: grid({ s: 10, m: 10, l: 10, xl: 10 }),
    };
  }
  return {
    className: grid12,
  };
})<TextWrapperProp>``;

const TitleWrapper = styled.div.attrs({
  className: classNames({
    'card-link__title': true,
    [font('wb', 4)]: true,
  }),
})``;

export const MediaObject: FunctionComponent<Props> = ({
  title,
  text,
  image,
}: Props): ReactElement<Props> => {
  const ImageComponent = image?.crops?.square && (
    <Image {...image.crops.square} />
  );

  const description = text && <PrismicHtmlBlock html={text} />;
  return (
    <MediaObjectBase
      url={null}
      title={title}
      Image={ImageComponent}
      partNumber={null}
      color={null}
      StatusIndicator={null}
      description={description}
      urlOverride={null}
      DateInfo={null}
      ExtraInfo={null}
      labels={{ labels: [] }}
      xOfY={{ x: null, y: null }}
      OverrideImageWrapper={ImageWrapper}
      OverrideTextWrapper={TextWrapper}
      OverrideTitleWrapper={TitleWrapper}
    />
  );
};

export default MediaObject;
