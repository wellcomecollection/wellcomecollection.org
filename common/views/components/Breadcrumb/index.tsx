import { FunctionComponent } from 'react';

import { BreadcrumbItems } from '@weco/common/model/breadcrumbs';
import { classNames, font } from '@weco/common/utils/classnames';
import { breadcrumbsLd } from '@weco/common/utils/json-ld';
import { links } from '@weco/common/views/components/Header';
import Space from '@weco/common/views/components/styled/Space';

export function getBreadcrumbItems(siteSection?: string): BreadcrumbItems {
  if (!siteSection) return { items: [] };
  return {
    items: siteSection
      ? [
          {
            text:
              links.find(link => link.siteSection === siteSection)?.title ||
              siteSection,
            url: siteSection ? `/${siteSection}` : '',
          },
        ]
      : [],
  };
}

const Breadcrumb: FunctionComponent<BreadcrumbItems> = ({
  items,
  noHomeLink,
}) => {
  // We prepend a 'Home' breadcrumb at the start of every chain, so every page
  // will ideally always have a visible breadcrumb.
  const allItems = [
    {
      text: 'Home',
      url: '/',
      isHidden: noHomeLink,
    },
    ...items,
  ];

  const visibleItems = allItems.filter(({ isHidden }) => !isHidden);

  if (visibleItems.length === 0) return null;

  return (
    // TODO remove is-hidden-print class once we've made the breadcrumbs more useful
    <div className="is-hidden-print" data-testid="breadcrumbs">
      {visibleItems.map(({ text, url, prefix }, i) => {
        const LinkOrSpanTag = url ? 'a' : 'span';

        return (
          <Space
            key={prefix ? `${prefix}-${text}` : text}
            as={prefix ? 'b' : 'span'}
            className={font('intr', 6)}
          >
            {i > 0 && (
              <Space
                as="span"
                $h={{ size: 's', properties: ['margin-left', 'margin-right'] }}
                aria-hidden="true"
              >
                |
              </Space>
            )}
            {prefix}{' '}
            <LinkOrSpanTag
              className={classNames({
                [font('intb', 6)]: Boolean(prefix),
              })}
              href={url}
              data-gtm-trigger={url ? 'breadcrumb_link' : undefined}
            >
              {text}
            </LinkOrSpanTag>
          </Space>
        );
      })}
      <script
        type="application/ld+json"
        dangerouslySetInnerHTML={{
          __html: JSON.stringify(breadcrumbsLd({ items })),
        }}
      />
    </div>
  );
};
export default Breadcrumb;
