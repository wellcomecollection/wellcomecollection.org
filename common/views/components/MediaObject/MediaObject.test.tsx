import MediaObject from './MediaObject';
import { shallowWithTheme } from '../../../test/fixtures/enzyme-helpers';
import { mockData } from '../../../test/fixtures/components/media-object';
import { mockImage } from '../../../test/fixtures/components/compact-card';
import { grid } from '../../../utils/classnames';

describe('MediaObject Component', () => {
  const component = shallowWithTheme(
    <MediaObject
      title={mockData.title}
      text={mockData.text}
      image={mockImage}
    />
  );
  const componentHtml = component.html();

  it('renders grid system of 2 for image and 10 for text', () => {
    expect(componentHtml.match(grid({ s: 2, m: 2, l: 2, xl: 2 }))).toBeTruthy();
    expect(
      componentHtml.match(grid({ s: 10, m: 10, l: 10, xl: 10 }))
    ).toBeTruthy();
  });
});
