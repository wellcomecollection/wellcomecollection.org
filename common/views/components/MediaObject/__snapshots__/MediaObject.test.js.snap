// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaObject Component should render snapshot 1`] = `
"<div class=\\"Space-sc-4az8w3-0 iRLDmT grid\\"><div class=\\"MediaObject__ImageWrapper-sc-1tzxyxh-0 grid__cell grid__cell--s2 grid__cell--m2 grid__cell--l2 grid__cell--xl2\\"><noscript>
      <img width='3200'
        height='3200'
        class='image image--noscript bg-charcoal font-white'
        src='https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&rect=397%2C0%2C2836%2C2836&w=640&h=640'
        alt='Two St. Bernard dogs with an avalanche victim, one tries to revive him while the other alerts the rescue party.' /></noscript><img width=\\"3200\\" height=\\"3200\\" class=\\"image bg-charcoal font-white lazy-image lazyload\\" src=\\"https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=30&amp;h=30\\" data-srcset=\\"https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=160&amp;h=160 160w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=180&amp;h=180 180w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=282&amp;h=282 282w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=320&amp;h=320 320w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=420&amp;h=420 420w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=600&amp;h=600 600w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=880&amp;h=880 880w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=960&amp;h=960 960w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=1024&amp;h=1024 1024w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=1338&amp;h=1338 1338w,https://images.prismic.io/wellcomecollection/c8f7b7f4-d455-4c3f-ae65-9dc7428ae60a_EP001430_0001.jpg?auto=compress%2Cformat&amp;rect=397%2C0%2C2836%2C2836&amp;w=2048&amp;h=2048 2048w\\" sizes=\\"100vw\\" alt=\\"Two St. Bernard dogs with an avalanche victim, one tries to revive him while the other alerts the rescue party.\\"/></div><div class=\\"MediaObject__TextWrapper-sc-1tzxyxh-1 grid__cell grid__cell--s10 grid__cell--m10 grid__cell--l10 grid__cell--xl10\\"><div class=\\"MediaObject__TitleWrapper-sc-1tzxyxh-2 card-link__title font-wb font-size-4 \\">Title</div><div class=\\"spaced-text font-hnl font-size-5 \\"><p>Lorem Ipsum is simply dummy text of the printing<a href=\\"https://www.gov.uk/government/publications/face-coverings-when-to-wear-one-and-how-to-make-your-own#when-you-do-not-need-to-wear-a-face-covering\\"> and t</a>ypesetting industry.</p></div></div></div>"
`;

exports[`MediaObject Component should render without an image 1`] = `"<div class=\\"Space-sc-4az8w3-0 iRLDmT grid\\"><div class=\\"MediaObject__ImageWrapper-sc-1tzxyxh-0 grid__cell grid__cell--s12 grid__cell--m12 grid__cell--l12 grid__cell--xl12\\"></div><div class=\\"MediaObject__TextWrapper-sc-1tzxyxh-1 grid__cell grid__cell--s12 grid__cell--m12 grid__cell--l12 grid__cell--xl12\\"><div class=\\"MediaObject__TitleWrapper-sc-1tzxyxh-2 card-link__title font-wb font-size-4 \\">Title</div><div class=\\"spaced-text font-hnl font-size-5 \\"><p>Lorem Ipsum is simply dummy text of the printing<a href=\\"https://www.gov.uk/government/publications/face-coverings-when-to-wear-one-and-how-to-make-your-own#when-you-do-not-need-to-wear-a-face-covering\\"> and t</a>ypesetting industry.</p></div></div></div>"`;
